src/Php80/bootstrap.php

```diff
@@ -26,13 +26,13 @@ function fdiv(float $dividend, float $divisor): float { return p\Php80::fdiv($di
     function preg_last_error_msg(): string { return p\Php80::preg_last_error_msg(); }
 }
 if (!function_exists('str_contains')) {
-    function str_contains(string $haystack, string $needle): bool { return p\Php80::str_contains($haystack, $needle); }
+    function str_contains(string $haystack = null, string $needle = null): bool { return p\Php80::str_contains($haystack, $needle); }
```

Comment:
How about we omit the type hint completely here. This way the parameter will be forced mandatory and the type check will occur in Php80 class only.

Reply:
Sounds good to me!

Reply:
that's not good. Having no typehint makes the behavior different for code running with strict_types and calling the polyfill (the typehint in the PHP80 class won't reproduce this, as they won't be called with strict types from this file)

Reply:
@stof raises a good point - [I've just checked what the behaviour of `str_contains` with `strict_types` is on PHP80](https://3v4l.org/B5eA1) and it would cause a `TypeError` if you pass null.

I don't think we resolve that in userland because we don't know if `strict_types` is on or not, either:

 * We allow the typehint to be nullable in the polyfill, and accept that the behaviour of `null` with `strict_types` differs from the PHP8 implementation
 * We don't allow the typehint to be nullable in the polyfill, and accept that the behaviour of `null` without `strict_types` differs from the PHP8 implementation

Reply:
Well, I'm OK if the behavior for `null` differs with strict_types (`null` is a special case for types of internal functions, and its behavior cannot be replicated in userland).
But removing the typehints here and keeping them only in the PHP80 class would mean that the behavior would *also* be different when passing other types (try passing an integer)

Reply:
Thanks @stof  - I've added unit tests for the various types, and kept the typehint. Are you happy with me resolving this?

(Sorry, I don't know the etiquette for resolving review conversations on Symfony PRs!)